### DISCUSSION TITLE
Object graph and IsValid

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#4289] [Questions] [answered]
We have a root object with a couple of Collections as child objects.
I implemented the solution recommended in #3438 
I could 'manually' find out if the child collection is valid but I would like something similar to single child objects e.g. For editable single child object, our root object 'automatically' knows the validity of child object. 

In our root object we do have the following method
```cs
      protected override void OnChildChanged(Csla.Core.ChildChangedEventArgs e)
       {
           if (e.ChildObject is Person)
           {
               BusinessRules.CheckRules(ObjPersonProperty);
               OnPropertyChanged(ObjPersonProperty);
           }
```

But calling the BusinessRules.CheckRules on collection child object is not marking the root object valid/invalid.

Kind Regards

**Comment by rockfordlhotka** [comment, marked as answer]
The `IsValid` property should return the value for the entire object graph. The `IsSelfValid` returns the value only for one object.

Collection objects are valid or not based on whether any of the child objects in the collection are valid.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka Yes Sir, you are right. I had made a mistake in my code.
